File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/converter/ContentItemConverter.java
package com.worldventures.dreamtrips.modules.trips.model.converter;

import com.worldventures.dreamtrips.api.trip.model.TripContent;
import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.modules.trips.model.ContentItem;

import io.techery.mappery.MapperyContext;

public class ContentItemConverter implements Converter<TripContent, ContentItem> {

   @Override
   public Class<TripContent> sourceClass() {
      return TripContent.class;
   }

   @Override
   public Class<ContentItem> targetClass() {
      return ContentItem.class;
   }

   @Override
   public ContentItem convert(MapperyContext mapperyContext, TripContent tripContent) {
      ContentItem contentItem = new ContentItem();
      contentItem.setName(tripContent.name());
      contentItem.setDescription(tripContent.description());
      contentItem.setLanguage(tripContent.language());
      contentItem.setTags(tripContent.tags());
      return contentItem;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/converter/ContentItemConverter.kt
package com.worldventures.dreamtrips.modules.trips.model.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.trip.model.TripContent
import com.worldventures.dreamtrips.modules.trips.model.ContentItem

import io.techery.mappery.MapperyContext

class ContentItemConverter : Converter<TripContent, ContentItem> {

   override fun sourceClass() = TripContent::class.java

   override fun targetClass() = ContentItem::class.java

   override fun convert(context: MapperyContext, source: TripContent) = ContentItem().apply {
      description = source.description()
      language = source.language()
      tags = source.tags()
      name = source.name()
   }
}
